tmf/org.eclipse.tracecompass.tmf.core/src/org/eclipse/tracecompass/tmf/core/config/TmfConfiguration.java

```diff
@@ -184,15 +290,79 @@ public Builder setParameters(Map<String, Object> parameters) {
          * @return a {@link ITmfConfiguration} instance
          */
         public ITmfConfiguration build() {
-            String typeId = fSourceTypeId;
-            if (typeId.isBlank()) {
-                throw new IllegalStateException("Configuration source type ID not set"); //$NON-NLS-1$
-            }
-            String id = fId;
-            if (id.isBlank()) {
-                throw new IllegalStateException("Configuration ID not set"); //$NON-NLS-1$
+            if (fId.isBlank()) {
+                fId = toUuidString(fName.isBlank() ? UNKNOWN : fName);
             }
             return new TmfConfiguration(this);
         }
     }
+    /**
+     * Converts a JSON parameters string to {@link ITmfConfiguration}
```

Comment:
The input is not a string. But I wonder if this method is still needed?

Reply:
Yes, it's not needed anymore. I'll remove it.